### DISCUSSION TITLE
Raspberry Pi error

### DISCUSSION THREAD

**Discussion by onnimikki** [ArthurSonzogni/Diagon#66] [General]
Hi,

Very cool project.  Am looking to use it with programming assignments that I make for my students.

Question: I'm trying to get it running on a raspberry pi.  Neither the snap nor the .deb packages worked, likely because it's an arm processor.  So I made it from source.  (Clear instructions!  Thanks!)

Running one of the examples generates an error:

```
$ diagon Math -- "1+1/2 + sum(i,0,10) = 112/2"
terminate called after throwing an instance of 'std::system_error'
  what():  Unknown error -1
Aborted
```

But simple executions like asking for the version is fine:

```
$ diagon --version
version: 156
Written by Arthur Sonzogni
License MIT
```

Any suggestion as to how to clear up the issue?

thanks!

James

**Comment by ArthurSonzogni** [comment]
Hi! Thanks for using Diagon! I would be curious how it get used.

About your issue, I tried today on Desktop. I observed the version build from source worked correctly. However the snap version got the exact same error.

I think you might have used the snap version, because I don't see a `./diagon` at the beginning, but `diagon`.

I will investigate today and get back to you.

**Comment by ArthurSonzogni** [comment]
I think I understand the issue.

I see this thread referenced the same issue and a fix:
https://github.com/antlr/antlr4/issues/3418#issuecomment-1451880253

I understand the problem, and I even proposed a fix in ANTLR:
https://github.com/antlr/antlr4/pull/4086

But, I though the fix was already merged ;-)
I will downgrade the ANTLR version for you, and push for the fix to land.

**Reply by ArthurSonzogni** [reply]
I landed the fix and update:
https://github.com/ArthurSonzogni/Diagon/commit/b2bf80fcaaec8774cbd630686f39da4aa716b3a0

I pushed the new version into the stable release of the snap package:

This should be automatic, but you can refresh it using:
```bash
sudo snap refresh diagon
```

Could you please let me know if your issue has been fixed?
